sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -206,8 +206,13 @@ object UnresolvedAttribute {
       val char = name(i)
       if (inBacktick) {
         if (char == '`') {
-          inBacktick = false
-          if (i + 1 < name.length && name(i + 1) != '.') throw e
+          if (i + 1 < name.length && name(i + 1) == '`') {
```

Comment:
maybe we should get rid of this hand-written parser and simply call `CatalystSqlParser.parseMultipartIdentifier` here.

Reply:
Thanks. I'll try to replace it with `CatalystSqlParser.parseMultipartIdentifier`.

Reply:
I found replacing with `CatlystSqlParser.parseMultipartIdentifier` breaks API compatibility.
If a string passed to `UnresolvedAttribute.parseAttributeName` contains quoted parts,  each of them is regarded as a complete name part.
Otherwise, each unquoted part is regarded as name parts which can be separated by `.` and each name part doesn't need to be quoted.

So, this is valid for `parseAttributeName` but not for `parseMultipartIdentifier`.
```
UnresolvedAttribute.parseAttributeName("*#  !.`abc`")
```